misc/ipv4_subnet.c

```diff
@@ -0,0 +1,60 @@
+#include <assert.h>
+#include <stdio.h>
+/**
+ * @file
+ * @brief
+ * [Explanation](https://www.softwaretestinghelp.com/subnet-mask-and-network-classes/)
+ * @author [XenoxJeager] (https://github.com/XenoxJeager)
+ * @details
+ * Getting subnetmasks of ipv4(32 bit long) addresses based on the length(0-32)
+ * of the network bits.
+ *
+ */
+
+/**
+ * @brief Definition of union ipv4_address composed of an unsigned int ipv4 and
+ * a four byte long char[] subnetmask.
+ */
+typedef union
+{
+    unsigned int ipv4;
+    unsigned char subnetmask[4];
```

Comment:
Please add a one-line description of what the variable is for (**see the example below**).

```c
int64_t data = 0;       ///< The value/key of the node.
int64_t array[64];      ///< The array that will be used for the tree.
```